tests/DownstreamRestApiTests/CredentialDescriptionTest.cs

```diff
@@ -0,0 +1,221 @@
+﻿using Microsoft.Identity.Abstractions;
+using System.Runtime.ConstrainedExecution;
+using Xunit;
+
+namespace UnitTests
+{
+    public class CredentialDescriptionTest
+    {
+        [Fact]
+        public void CredentialDescriptionProperties()
```

Comment:
This needs to be theory based.
Each test case needs to validated on its own.
This model of multiple test cases leads to 'fix one, then the next'.

Reply:
will do
I was tempted to use t4, but I wasn't sure you'd have liked it. Theories are similiar in spirit